api/apps/v1alpha1/nemo_evaluator_types.go

```diff
@@ -0,0 +1,764 @@
+}
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.state`,priority=0
+// +kubebuilder:printcolumn:name="Age",type="date",format="date-time",JSONPath=".metadata.creationTimestamp",priority=0
+
+// NemoEvaluator is the Schema for the NemoEvaluator API
+type NemoEvaluator struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NemoEvaluatorSpec   `json:"spec,omitempty"`
+	Status NemoEvaluatorStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// NemoEvaluatorList contains a list of NemoEvaluator
+type NemoEvaluatorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NemoEvaluator `json:"items"`
+}
+
+type Mongodb struct {
+	Endpoint string `json:"endpoint"`
+}
+
+type ArgoWorkFlows struct {
+	Endpoint       string `json:"endpoint"`
+	ServiceAccount string `json:"serviceAccount"`
+}
+
+type Milvus struct {
+	Endpoint string `json:"endpoint"`
+}
+
+type DataStore struct {
+	Endpoint string `json:"endpoint"`
+}
+
+// GetStandardSelectorLabels returns the standard selector labels for the NemoEvaluator deployment
+func (n *NemoEvaluator) GetStandardSelectorLabels() map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name": n.Name,
+	}
+}
+
+// GetStandardLabels returns the standard set of labels for NemoEvaluator resources
+func (n *NemoEvaluator) GetStandardLabels() map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name":             n.Name,
+		"app.kubernetes.io/instance":         n.Name,
+		"app.kubernetes.io/operator-version": os.Getenv("OPERATOR_VERSION"),
+		"app.kubernetes.io/part-of":          "nemo-evaluator-service",
+		"app.kubernetes.io/managed-by":       "k8s-nim-operator",
+	}
+}
+
+// GetStandardEnv returns the standard set of env variables for the NemoEvaluator container
+func (n *NemoEvaluator) GetStandardEnv() []corev1.EnvVar {
+	// add standard env required for NIM service
+	envVars := []corev1.EnvVar{
+		{
+			Name:  "NAMESPACE",
+			Value: n.Namespace,
+		},
+		{
+			Name: "HOST_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "status.hostIP",
+				},
+			},
+		},
+		{
+			Name:  "EVALUATOR_HOST",
+			Value: "0.0.0.0",
+		},
+		{
+			Name:  "EVALUATOR_PORT",
+			Value: "7331",
+		},
+		{
+			Name:  "DATABASE_URI",
+			Value: n.Spec.Mongodb.Endpoint,
+		},
+		{
+			Name:  "DATABASE_NAME",
+			Value: "evaluations",
+		},
+		{
+			Name:  "ARGO_HOST",
+			Value: n.Spec.ArgoWorkFlows.Endpoint,
+		},
+		{
+			Name:  "MILVUS_URL",
+			Value: n.Spec.Milvus.Endpoint,
+		},
+		{
+			Name:  "SERVICE_ACCOUNT",
+			Value: n.Spec.ArgoWorkFlows.ServiceAccount,
+		},
+		{
+			Name:  "DATA_STORE_HOST",
+			Value: n.Spec.DataStore.Endpoint,
+		},
+		{
+			Name: "INFERENCE_URL",
```

Comment:
does this need to be the NIM endpoint that we deployed? we also need to define a param to get this input.